### ISSUE TITLE
Document that Module.run() accepts an array, in addition to a function

### ISSUE THREAD

**Issue by jyasskin** [angular/angular.js#11399]
Reading https://docs.angularjs.org/api/ng/type/angular.Module#run gives no indication that the argument can be either an array or a function. https://docs.angularjs.org/guide/di#module-methods describes it, but that's hard to find when you've looked up `run()` to understand the arguments you're seeing in some existing code.

**Comment by wesleycho** [comment]
None of the other methods contain a mention to the array syntax for injection either.

Are you proposing that all of them should have it mentioned in that angular.module doc page?  I'm not sure whether the methods need it documented there, but a little blurb about dependency injection at the top of the page with a link to the guide's page on DI and an example of using the optional array syntax for injection probably would help.

**Comment by jyasskin** [comment]
Yes, sorry for picking out just the one I happened to run into. I think there should be some way to get from each definition to a description of the way an `Array` is allowed as a parameter. I would have found the answer faster if it were at the top of the `angular.Module` doc page, but I would still have had to hunt. That would also leave out any functions in other interfaces that happen to accept the same type (perhaps to forward it to a `Module` method).

I notice that the table describing the `initializationFn` parameter has a link on the `Function` type that doesn't go anywhere. If you defined a typedef for this injectable function type, would that let you document it all in one place?

**Comment by trevorade** [comment]
The fact is that the documentation is incorrect.  It is supported to pass in an array but not documented.  This becomes important for compiled code (production code as well as tests).
